controllers/storage/minio.go

```diff
@@ -65,19 +64,20 @@ var (
 // Reconciler implements the reconcile logic of minIO service
 func (m *MinIOReconciler) Reconcile() (*lcm.CRStatus, error) {
 	var minioCR minio.MinIOInstance
-
 	if m.HarborCluster.Spec.Storage.Kind != inClusterStorage {
-		var exSecret *corev1.Secret
-		err := m.KubeClient.Get(m.getExternalSecretNamespacedName(), exSecret)
+		var exSecret corev1.Secret
+		err := m.KubeClient.Get(m.getExternalSecretNamespacedName(), &exSecret)
 		if k8serror.IsNotFound(err) {
 			return m.ProvisionExternalStorage()
 		} else if err != nil {
+			m.Log.Error(err, "failed to get secret", m.getExternalSecretNamespacedName())
 			return minioNotReadyStatus(GetExternalSecretError, err.Error()), err
 		}
 
-		m.CurrentExternalSecret = exSecret
+		m.CurrentExternalSecret = &exSecret
 		m.DesiredExternalSecret, err = m.generateExternalSecret()
 		if err != nil {
+			m.Log.Error(err, "failed to generate external secret", m.getExternalSecretNamespacedName())
```

Comment:
The returned error of `Reconcile()` must be logged at the upper calling level, why we still need to log it here?